core/controllers/creator_dashboard_test.py

```diff
@@ -644,6 +697,120 @@ def test_can_see_subscribers(self):
         response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
         self.assertEqual(len(response['subscribers_list']), 0)
 
+    def test_get_topic_summary_dicts_with_new_structure_players_enabled(self):
+        self.login(self.OWNER_EMAIL)
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
+            self.assertEqual(len(response['topic_summary_dicts']), 0)
+            self.save_new_topic(
+                'topic_id', self.owner_id, 'Name', 'Description',
+                ['story_id_1', 'story_id_2'], ['story_id_3'],
+                ['skill_id_1', 'skill_id_2'], [], 1)
+            response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
+            self.assertEqual(len(response['topic_summary_dicts']), 1)
+            self.assertTrue(isinstance(response['topic_summary_dicts'], list))
+            self.assertEqual(response['topic_summary_dicts'][0]['name'], 'Name')
+            self.assertEqual(
+                response['topic_summary_dicts'][0]['id'], 'topic_id')
+        self.logout()
+
+    def test_get_no_topic_summary_dicts_with_new_structure_players_disabled(
+            self):
+        self.login(self.OWNER_EMAIL)
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', False):
+            response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
+            self.assertIsNone(response.get('topic_summary_dicts'))
+        self.logout()
+
+    def test_can_update_display_preference(self):
+        self.login(self.OWNER_EMAIL)
+        display_preference = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['display_preference']
+        self.assertEqual(display_preference, 'card')
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.post_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL,
+            {'display_preference': 'list'},
+            csrf_token=csrf_token)
+        display_preference = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['display_preference']
+        self.assertEqual(display_preference, 'list')
+        self.logout()
+
+    def test_can_create_collections(self):
+
+        class MockCollection(object):
+            @classmethod
+            def mock_create_default_collection(cls, collection_id):
+                """Mocks create_default_collection()."""
+                return collection_domain.Collection(
+                    collection_id, 'A title', 'A category',
+                    'An objective', constants.DEFAULT_LANGUAGE_CODE, [],
+                    feconf.CURRENT_COLLECTION_SCHEMA_VERSION, [], 0)
+
+        with self.swap(
+            collection_domain.Collection, 'create_default_collection',
+            MockCollection.mock_create_default_collection):
```

Comment:
Nope -- drop this, use the actual post_json() to the handler instead without mocking anything.

Otherwise you aren't actually testing Collection.create_default_collection, which seems to defeat the point somewhat.

Reply:
Done but you suggested we should update the collection to have useful data to test against in the first place [here](https://github.com/oppia/oppia/pull/6480#discussion_r279293395)

Reply:
Yes, but that was when your test seemed to be about testing a different thing. That comment still stands, and you did address it successfully in test_get_collections_list(), so it's all good.

This test doesn't test that though; it tests a new collection, and that is already a strictly-defined concept. So we should use that.

I hope this clarifies!